src/init.cpp

```diff
@@ -1553,19 +1547,16 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
     nTotalCache = std::min(nTotalCache, nMaxDbCache << 20); // total cache cannot be greated than nMaxDbcache
     int64_t nBlockTreeDBCache = nTotalCache / 8;
 
-#ifdef ENABLE_ADDRESS_INDEXING
+
     if (GetBoolArg("-addressindex", DEFAULT_ADDRESSINDEX) || GetBoolArg("-spentindex", DEFAULT_SPENTINDEX)) {
         // enable 3/4 of the cache if addressindex and/or spentindex is enabled
         nBlockTreeDBCache = nTotalCache * 3 / 4;
     } else {
-#endif // ENABLE_ADDRESS_INDEXING
         if (nBlockTreeDBCache > (1 << 21) && !GetBoolArg("-txindex", false)) {
```

Comment:
The default value is missing, this should be `GetBoolArg("-txindex", DEFAULT_TXINDEX)`.

Reply:
Addressed

Reply:
Same modification is needed at line 1029.

Reply:
This makes me question the whole logic behind `GetBoolArg()` and the defaults associated like DEFAULT_TXINDEX
All command line arguments parsed by GetBoolArg() are **flags** which mean `true` when explicitly defined and `false` when missing. Hence setting a default constant for those flags has no meaning and only adds unneded verbosity to code.

As those flags are options, should we change for instance at a certain point DEFAULT_TXINDEX to true it would mean the user not wanting that index should be forced to set `-txindex=false` which means forcing him to explictly deny the enforcement of an option (which is against the logic of how an option is perceived).

In case txindex should become enabled by default then we should change also command line arguments to `-wno-txindex` to optionally disable the feature.

Reply:
Looking at the implementation of GetBoolArg() (util.cpp ~line 397) you see default is returned **only** if the command argument is not present and, as a result, being an **option** the default cannot be anything else than `false`

Reply:
Reading further things are even worse.
While parsing arguments (from CLI or Config file) at the very end the so called negative settings (i.e. a setting name prefixed by `-no` .e.g `-notxindex`) are parsed too but in a very clumsy and ineffective way.
The negation arg is effective only if the positive flag is **NOT** set which leads to the uncomfortable situation where the user can enter something like `./zend -txindex -notxindex` maybe expecting the indexing is disabled which instead is not true. 🤕